### PR TITLE
Main phase1 integration collect note group command

### PR DESCRIPTION
**Pull request by EduardCoretchi**

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -32,18 +32,39 @@ public void robotInit() {
         shooterSystem = new ShooterSystem();
         intakeSystem = new IntakeSystem();
         armSystem = new ArmSystem();
-        xboxController = new XboxController(0);
+        xboxController = new XboxController(RobotMap.XBOX_CONTROLLER_PORT);
 
-        POVButton dPadUp = new POVButton(xboxController, 0);
-        POVButton dPadDown = new POVButton(xboxController, 180);
+        armSystem.setDefaultCommand(
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java:40]: So basically you are always running a motion to the shooter, and while doing so you are moving the intake.
why is this a default command?

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -32,18 +32,39 @@ public void robotInit() {
...
+
+        POVButton dPadUp = new POVButton(xboxController, RobotMap.D_PAD_TO_SHOOTER_ANGLE);
+        POVButton dPadDown = new POVButton(xboxController, RobotMap.D_PAD_TO_FLOOR_ANGLE);
 
         dPadUp.onTrue(new ArmMoveToShooterCommand(armSystem));
         dPadDown.onTrue(new ArmMoveToFloorCommand(armSystem));
 
         new JoystickButton(xboxController, XboxController.Button.kX.value).whileTrue(new ShootOut(shooterSystem));
-        new JoystickButton(xboxController, XboxController.Button.kB.value).whileTrue(new ShooterPID(shooterSystem, 2000));
+        //new JoystickButton(xboxController, XboxController.Button.kB.value).whileTrue(new ShooterPID(shooterSystem, RobotMap.SHOOTER_TARGET_RPM));
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: either uncomment or remove. Don't keep commented code.

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -32,18 +32,39 @@ public void robotInit() {
...
         new JoystickButton(xboxController, XboxController.Button.kY.value).whileTrue(new OuttakeCommand(intakeSystem));
-        new JoystickButton(xboxController, XboxController.Button.kA.value).whileTrue(new IntakeCommand(intakeSystem));
+        //new JoystickButton(xboxController, XboxController.Button.kA.value).whileTrue(new IntakeCommand(intakeSystem));
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: either uncomment or remove. Don't keep commented code.

File: src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -32,18 +32,39 @@ public void robotInit() {
...
+
+        ParallelDeadlineGroup collectNote = new ParallelDeadlineGroup(
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: I guess this will work, considering that a note will only enter when the arm is on the floor. But maybe first reach the floor with the arm and only then start `IntakeCommand`?

File: src/main/java/frc/robot/commands/IntakeCommand.java [hunk 5]
```diff
@@ -24,6 +24,6 @@ public void end(boolean interrupted) {
     }
 
     public boolean isFinished() {
-        return intakeSystem.isNoteInside();
+        return false;
```
Comment (tomtzook) [inline src/main/java/frc/robot/commands/IntakeCommand.java:27]: why did you remove the condition?

File: src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java [hunk 6]
```diff
@@ -24,8 +24,9 @@ public void initialize() {
     @Override
     public void execute() {
         double output = pidController.calculate(armSystem.getAbsEncoderPositionDegrees(),  RobotMap.ARM_SHOOTER_ANGLE);
+        pidController.setIZone(RobotMap.ARM_PID_I_ZONE);
```
Comment (tomtzook) [inline src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java]: don't change Izone during calculation,  set it in the ctor once.

File: src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java [hunk 7]
```diff
@@ -24,8 +24,9 @@ public void initialize() {
...
 
-        armSystem.move(output);
+        armSystem.move(output + RobotMap.ARM_PID_K_GRAVITY * Math.cos(Math.toRadians(armSystem.getAbsEncoderPositionDegrees())));
```
Comment (tomtzook) [inline src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java:29]: is this addition of feed-forward necessary and tested?

File: src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java [hunk 8]
```diff
@@ -24,8 +24,9 @@ public void initialize() {
     @Override
     public void execute() {
         double power = pidController.calculate(armSystem.getAbsEncoderPositionDegrees(), RobotMap.ARM_FLOOR_ANGLE);
+        pidController.setIZone(RobotMap.ARM_PID_I_ZONE);
```
Comment (tomtzook) [inline src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java]: don't change Izone during calculation, set it in the ctor once.

File: src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java [hunk 9]
```diff
@@ -24,8 +24,9 @@ public void initialize() {
...
 
-        armSystem.move(power);
+        armSystem.move(power + RobotMap.ARM_PID_K_GRAVITY * Math.cos(Math.toRadians(armSystem.getAbsEncoderPositionDegrees())));
```
Comment (tomtzook) [inline src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java:29]: is this addition of feed-forward necessary and tested?

**Review by tomtzook — CHANGES_REQUESTED** [review]
saw some changes that seem unrelated to adding the command group.

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/Robot.java:40]: This is created for the purpose to simplify other code instead of write this part of code in other command group.

File: src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/Robot.java]: done

File: src/main/java/frc/robot/Robot.java (hunk 3 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/Robot.java]: done

File: src/main/java/frc/robot/Robot.java (hunk 4 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/Robot.java]: done

File: src/main/java/frc/robot/commands/IntakeCommand.java (hunk 5 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/commands/IntakeCommand.java:27]: for making possible to use the IntakeCommand and move the intake wheels with a note in intake

File: src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java (hunk 6 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java]: done

File: src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java (hunk 8 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java]: done

File: src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java (hunk 7 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java:29]: yep, and it worked nicely

File: src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java (hunk 9 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java:29]: yep, and it work nicely

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (tomtzook) [reply src/main/java/frc/robot/Robot.java:40]: I see, but what will realistically happen is that this command starts running, moves the arm to shooter and then ends. Afterwards it will start again (because no command is running on the arm) and will start moving and stop a few milliseconds after, and so on and so on. We may need a different solution.

Luckily command groups hold the requirements used within them for the duration so this will not fuck up command groups using the arm

File: src/main/java/frc/robot/Robot.java [hunk 10]
```diff
@@ -32,18 +32,39 @@ public void robotInit() {
...
-        xboxController = new XboxController(0);
+        xboxController = new XboxController(RobotMap.XBOX_CONTROLLER_PORT);
 
-        POVButton dPadUp = new POVButton(xboxController, 0);
-        POVButton dPadDown = new POVButton(xboxController, 180);
+        armSystem.setDefaultCommand(
+                new ParallelCommandGroup(
+                        new ParallelDeadlineGroup(
+                                new WaitUntilCommand(() -> armSystem.reachedATargetAngle(RobotMap.ARM_SHOOTER_ANGLE)),
+                                new InstantCommand(() -> intakeSystem.slowIn())
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: this is useful for any arm-to-shooter motion, including other command groups, we will need to share this between them. But I guess it can wait for later

File: src/main/java/frc/robot/Robot.java (hunk 4 above)
Comment (tomtzook) [reply src/main/java/frc/robot/Robot.java]: The command runs first the Intake and them the arm to the floor. Should be arm-to-floor and then intake

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/Robot.java:40]: But the default command doesn't end. He wait that ArmMoveToShooterCommand end, but he doesn't have 'isFinished' so he going to work for endless

File: src/main/java/frc/robot/Robot.java (hunk 4 above)
Comment (EduardCoretchi) [reply src/main/java/frc/robot/Robot.java]: Done

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (tomtzook) [reply src/main/java/frc/robot/Robot.java:40]: my mistake

**Comment by tomtzook** [comment]
We just need to test the behavior of the default command and then this will be approved.

**Review by tomtzook — APPROVED** [review]
